src/EnergyPlus/UnitarySystem.cc

```diff
@@ -7591,9 +7615,8 @@ namespace UnitarySystems {
 
         // check supplemental heating coil outlet temp based on maximum allowed
         if (this->m_SuppCoilExists) {
-            SuppPLR = this->m_SuppHeatPartLoadFrac;
             // only need to test for high supply air temp if supplemental coil is operating
-            if (SuppPLR > 0.0) {
+            if (SupHeaterLoad > 0.0 && this->m_DesignSuppHeatingCapacity > 0.0) {
```

Comment:
This change might be why there are diffs. I would revert this logic unless there is good reason for it. The cooling and heating coil are modeled first to determine the air flow rate and PLRs. If after that the supp heater is needed, then SuppPLR will be > 0 and the supp heater will turn on. Also, if (... && this->m_DesignSuppHeatingCapacity > 0.0), of course it is.

Reply:
Changes to this part are needed to allow multistage backup coils, but I will try reverting the logic to be as close to previous one as possible for other cases.

Reply:
You could use something like this (here and other places) just to get it to work and we can figure out better logic later. There should be no diffs from adding a coil that doesn't exist yet.

    if (this->m_SuppCoilExists) {
        if( stagedheatingcoil) {
            insert your logic
        } else {
           existing code
        }
    }